tests/ui/consts/const-eval/validate_uninhabited_zsts.64bit.stderr

```diff
@@ -11,7 +11,7 @@ error[E0080]: evaluation of constant value failed
   --> $DIR/validate_uninhabited_zsts.rs:4:14
    |
 LL |     unsafe { std::mem::transmute(()) }
-   |              ^^^^^^^^^^^^^^^^^^^^^^^ transmuting to uninhabited type
+   |              ^^^^^^^^^^^^^^^^^^^^^^^ entering unreachable code
```

Comment:
Any thoughts on the best lowering to use for this, to be able to keep a nice message?

I was considering an always-fails `TerminatorKind::Assert`, but that needs a target `BasicBlock`.  Similar, I considered a `match foo {}`-style `SwitchInt` on the argument, but that *also* needs at least an `otherwise` target.

Or maybe the "unreachable" message is ok, since these usually get a nice lint anyway (like the one earlier in this file).

Reply:
Oh, dereferencing an uninhabited pointer also give "entering unreachable code":

https://github.com/rust-lang/rust/blob/34e6673a0473e90ef01a18eb575392c9e3859747/src/tools/miri/tests/fail/never_say_never.stderr#L1-L6

So doing the same here seems at least plausible.

Reply:
Yeah, imo this is fine